SingularityBase/src/main/java/com/hubspot/singularity/SingularityDeployProgress.java

```diff
@@ -56,21 +66,22 @@ public long getTimestamp() {
   }
 
   public SingularityDeployProgress withNewInstances(int instances) {
-    return new SingularityDeployProgress(instances, deployInstanceCountPerStep, deployStepWaitTimeMs, false, autoAdvanceDeploySteps, failedDeployTasks, System.currentTimeMillis());
+    return new SingularityDeployProgress(instances, currentActiveInstances, deployInstanceCountPerStep, deployStepWaitTimeMs, false, autoAdvanceDeploySteps, failedDeployTasks, System.currentTimeMillis());
   }
 
   public SingularityDeployProgress withCompletedStep() {
-    return new SingularityDeployProgress(targetActiveInstances, deployInstanceCountPerStep, deployStepWaitTimeMs, true, autoAdvanceDeploySteps, failedDeployTasks, System.currentTimeMillis());
+    return new SingularityDeployProgress(targetActiveInstances, currentActiveInstances + 1, deployInstanceCountPerStep, deployStepWaitTimeMs, true, autoAdvanceDeploySteps, failedDeployTasks, System.currentTimeMillis());
```

Comment:
This was originally intended to just be the current deploy progress with `stepCompleted` flipped to true. If we want to update the active instance count I'd suggest getting it from the `deployActiveTasks` variable that gets passed around instead. If we pass that to the calls of `markStepFinished`, we can construct the new deploy progress with the actual count of active tasks after each step. `withCompletedStep` should do just what I mentioned above and flip the value of that bool, keeping the rest the same